File: app/src/main/java/lib/phonograph/preference/DialogPreferenceX.java
/*
 * Copyright (c) 2022 chr_56 & Abou Zeid (kabouzeid) (original author)
 */

package lib.phonograph.preference;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.DialogPreference;

import player.phonograph.R;

/**
 * @author Aidan Follestad (afollestad)
 */
public class DialogPreferenceX extends DialogPreference {

    public DialogPreferenceX(Context context) {
        super(context);
        init(context, null);
    }

    public DialogPreferenceX(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public DialogPreferenceX(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    public DialogPreferenceX(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        setLayoutResource(R.layout.x_preference);
    }
}

File: app/src/main/java/lib/phonograph/preference/DialogPreferenceX.kt
/*
 * Copyright (c) 2022 chr_56 & Abou Zeid (kabouzeid) (original author)
 */
package lib.phonograph.preference

import android.content.Context
import android.util.AttributeSet
import androidx.preference.DialogPreference
import player.phonograph.R

/**
 * @author Aidan Follestad (afollestad)
 */
class DialogPreferenceX : DialogPreference {
    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context,
        attrs,
        defStyleAttr,
        defStyleRes) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        layoutResource = R.layout.x_preference
    }
}